### PR TITLE
selinux: allow bulk deletion of selinux notifications

### LINKED ISSUE #14033: bulk dismiss selinux alerts
It would be nice to have the ability to bulk-dismiss SELinux alerts, this would be particularly helpful when building policies and wanting to wipe the cached entries clean to see what new ones appear.

### PR DESCRIPTION
**Pull request by KKoukiou**
Fixes #14033

image

### CODE REVIEW

File: pkg/selinux/setroubleshoot-view.jsx [hunk 1]
```diff
@@ -386,8 +386,6 @@ export class SETroubleshootPage extends React.Component {
                     { title: itm.description }
                 ];
                 if (itm.count > 1) {
-                    title = cockpit.format(cockpit.ngettext("$0 occurrence", "$0 occurrences", itm.count),
-                                           itm.count);
```
Comment (martinpitt) [inline pkg/selinux/setroubleshoot-view.jsx:390]: It's being used in line 406 below in the `<CardTitle>`. Indeed it makes not much sense to change it to *this* string, as the default is "SELinux access control errors". But I feel this requires a little more research what this was really meant to do?

**Review by martinpitt — COMMENTED** [review]
I'll do a proper review tomorrow, just one quick remark

**Comment by KKoukiou** [comment]
@martinpitt thanks for the thoughts and insight on the design! I totally agree with your remarks. Your suggestion sounds good solution indeed!  Let me give it a try :)

File: pkg/selinux/setroubleshoot-view.jsx (hunk 1 above)
Comment (KKoukiou) [reply pkg/selinux/setroubleshoot-view.jsx:390]: Explained it in the commit message what went wrong - found the falty commit in the git history.

**Comment by martinpitt** [comment]
Yay, this is much nicer IMHO:

image

I'll make a screenshot with two selections for the release notes, put it into the description. Thanks!

File: pkg/selinux/setroubleshoot-view.jsx [hunk 2]
```diff
@@ -318,6 +310,12 @@ export class SETroubleshootPage extends React.Component {
         e.stopPropagation();
     }
 
+    onSelect(_, isSelected, rowId) {
+        const selected = Object.assign(this.state.selected);
+        selected[this.props.entries[rowId / 2].key] = isSelected;
```
Comment (martinpitt) [inline pkg/selinux/setroubleshoot-view.jsx:315]: Divisions in an array index, and doing arithmetic with an ID both look weird..  That could at least use a comment. There is somehow two rows for an entry? Can we be sure the ID is even? Can we even be sure that it contains a number? (type-wise, it's a string, no?) Could it be `//` to make sure it uses integer division, or perhaps avoid arithmetic at all?

Comment (KKoukiou) [reply pkg/selinux/setroubleshoot-view.jsx:315]: There is a row also for the expandable items. Therefore it's tr = number of rows * 2.

onSelect returns an index, not an DOM id. That's a confusing name for this variable indeed. 

For avoiding arithmetic at all I need to rewrite cockpit-components-table.jsx to use the new PF4 table (this uses an deprecated PF4 table) which I want to do - but out of the scope of this PR.

Comment (martinpitt) [reply pkg/selinux/setroubleshoot-view.jsx:315]: Thanks for the explanation -- so this will disappear eventually, good.

**Review by martinpitt — APPROVED** [review]

**Pull request merged**